modules/rubiconAnalyticsAdapter.js

```diff
@@ -0,0 +1,441 @@
+import adapter from 'src/AnalyticsAdapter';
+import adaptermanager from 'src/adaptermanager';
+import CONSTANTS from 'src/constants.json';
+import { ajax } from 'src/ajax';
+import { config } from 'src/config';
+import * as utils from 'src/utils';
+
+const {
+  EVENTS: {
+    AUCTION_INIT,
+    AUCTION_END,
+    BID_REQUESTED,
+    BID_RESPONSE,
+    BIDDER_DONE,
+    BID_TIMEOUT,
+    BID_WON,
+    SET_TARGETING
+  },
+  STATUS: {
+    GOOD,
+    NO_BID
+  }
+} = CONSTANTS;
+
+let serverConfig;
+config.getConfig('s2sConfig', ({s2sConfig}) => {
+  serverConfig = s2sConfig;
+});
+
+const ENDPOINT = '//localhost:3000/event';
+export const SEND_TIMEOUT = 3000;
+const INTEGRATION = 'pbjs';
+
+const cache = {
+  auctions: {},
+  targeting: {},
+  timeouts: {},
+};
+
+// basically lodash#pick that also allows transformation functions and property renaming
+function _pick(obj, properties) {
```

Comment:
Should we move this to utils ?

Reply:
We could, but I don't think it's necessary.  I'd rather leave it here until such time that someone else decides they need to heavily utilize a pick function.  Plus I deviated a bit from lodash's implementation of `pick`, which could be confusing for others to use if they're not familiar with my intent.